tests/loader_test_vectors.rs

```diff
@@ -0,0 +1,231 @@
+    "ion-tests/iontestdata/good/utf32.ion",
+    "ion-tests/iontestdata/good/subfieldVarUInt32bit.ion",
+    "ion-tests/iontestdata/good/typecodes/T6-large.10n",
+    "ion-tests/iontestdata/good/typecodes/T7-large.10n",
+    "ion-tests/iontestdata/good/typecodes/type_6_length_0.10n",
+    // these appear to have a problem specific to how we're calling ion-c (amzn/ion-rust#218)
+    "ion-tests/iontestdata/good/equivs/intsLargeNegative1.10n",
+    "ion-tests/iontestdata/good/equivs/intsLargePositive1.10n",
+    "ion-tests/iontestdata/good/intLongMaxValuePlusOne.10n",
+    "ion-tests/iontestdata/good/item1.10n",
+    "ion-tests/iontestdata/good/subfieldVarInt.ion",
+    "ion-tests/iontestdata/good/typecodes/T2.10n",
+    "ion-tests/iontestdata/good/typecodes/T3.10n",
+    // these are symbols with unknown text (amzn/ion-rust#219)
+    "ion-tests/iontestdata/good/symbolExplicitZero.10n",
+    "ion-tests/iontestdata/good/symbolImplicitZero.10n",
+    "ion-tests/iontestdata/good/symbolZero.ion",
+    "ion-tests/iontestdata/good/typecodes/T7-small.10n",
+];
+
+/// Files that should only be skipped in equivalence file testing
+const EQUIVS_SKIP_LIST: &[&str] = &[
+    // ion-c seems to have a problem with negative binary literals (amzn/ion-c#235)
+    "ion-tests/iontestdata/good/equivs/binaryInts.ion",
+    "ion-tests/iontestdata/good/equivs/intsWithUnderscores.ion",
+];
+
+/// Files that should only be skipped in non-equivalence file testing
+const NON_EQUIVS_SKIP_LIST: &[&str] = &[
+    // we need a structural equality (IonEq) for these (amzn/ion-rust#220)
+    "ion-tests/iontestdata/good/non-equivs/decimals.ion",
+    "ion-tests/iontestdata/good/non-equivs/floatsVsDecimals.ion",
+    "ion-tests/iontestdata/good/non-equivs/floats.ion",
+    // these have symbols with unknown text (amzn/ion-rust#219)
+    "ion-tests/iontestdata/good/non-equivs/symbolTablesUnknownText.ion",
+    // this is a bug in our PartialEq (amzn/ion-rust#216)
+    "ion-tests/iontestdata/good/non-equivs/structs.ion",
+];
+
+/// Concatenates two slices of string slices together.
+#[inline]
+fn concat<'a>(left: &[&'a str], right: &[&'a str]) -> Vec<&'a str> {
+    left.into_iter()
+        .chain(right.into_iter())
+        .map(|p| *p)
+        .collect()
+}
+
+fn load_file<L: Loader>(loader: &L, file_name: &str) -> IonResult<Vec<OwnedElement>> {
+    // TODO have a better API that doesn't require buffering into memory everything...
+    let data = read(file_name)?;
+    loader.load_all(&data)
+}
+
+fn assert_file<T, F: FnOnce() -> IonResult<T>>(skip_list: &[&str], file_name: &str, asserter: F) {
+    if skip_list
+        .into_iter()
+        // TODO construct the paths in a not so hacky way
+        .map(|p| p.replace("/", &PATH_SEPARATOR.to_string()))
+        .find(|p| p == file_name)
+        .is_some()
+    {
+        println!("IGNORING: {}", file_name);
+    } else {
+        println!("TESTING: {}", file_name);
+        asserter().unwrap();
+    }
```

Comment:
It would be nice to `#[ignore]` these, but the crate we're pulling in doesn't support that (there is a very stale PR to add this, but the author of the crate has not commented).  If we care to, we might want to fork the crate for our purposes.